File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务层
app.service('brandService',function($http){

    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../brand/findAll.do');
    }
    //分页
    this.findPage=function(page,rows){
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    }
    //查询实体
    this.findTbBrandById=function(id){
        return $http.get('../brand/findTbBrandById.do?id='+id);
    }
    //增加
    this.addTbBrand=function(entity){
        return  $http.post('../brand/addTbBrand.do',entity );
    }
    //修改
    this.updateTbBrand=function(entity){
        return  $http.post('../brand/updateTbBrand.do',entity );
    }
    //删除
    this.dele=function(ids){
        return $http.get('../brand/deleteTbBrand.do?ids='+ids);
    }
    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../brand/SerchPage.do?page='+page+"&rows="+rows, searchEntity);
    }
    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }

});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
	/**
	 * 查找全部
	 *
	 * @return
	 */
	public List<TbBrand> findAll();

	/**
	 * 分页查询
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 添加品牌
	 */
	public void addTbBrand(TbBrand tbBrand);

	/**
	 * 修改品牌
	 *
	 * @param tbBrand
	 */
	public void updateTbBrand(TbBrand tbBrand);

	/**
	 * 通过ID 获取当前实体 然后回填到修改的框中
	 *
	 * @param id
	 * @return
	 */
	public TbBrand findTbBrandById(Long id);

	/**
	 * 删除品牌
	 *
	 * @param ids
	 */
	public void deleteTbBrand(Long[] ids);


	/**
	 * @param tbBrand
	 * @param pageNums
	 * @param pageSize
	 * @return
	 */
	public PageResult SerchPage(TbBrand tbBrand, int pageNums, int pageSize);


	/**
	 * 品牌下拉框数据
	 */
	List<Map> selectOptionList();

}
